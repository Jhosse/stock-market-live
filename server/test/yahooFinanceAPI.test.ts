import { gql } from 'apollo-server-express';
import { query } from './setupServer'

describe('Yahoo Finance API', () => {
  it('summaryAndSpark', async () => {

    const res = await query({
      query: gql`
        query {
          summaryAndSpark {
            marketSummaryAndSparkResponse {
              result {
                symbol
              }
            }
          }
        }
      `
    })

    expect(res.data).toBeDefined()
    expect(res.errors).toBeUndefined()
    expect(res.data).toMatchSnapshot()
  })
})

