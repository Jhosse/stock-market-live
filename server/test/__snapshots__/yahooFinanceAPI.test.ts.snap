// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Yahoo Finance API summaryAndSpark 1`] = `
Object {
  "summaryAndSpark": Object {
    "marketSummaryAndSparkResponse": Object {
      "result": Array [
        Object {
          "symbol": "^FTSE",
        },
        Object {
          "symbol": "^FTMC",
        },
        Object {
          "symbol": "^FTAI",
        },
        Object {
          "symbol": "GBPEUR=X",
        },
        Object {
          "symbol": "GBPUSD=X",
        },
        Object {
          "symbol": "BTC-GBP",
        },
        Object {
          "symbol": "^CMC200",
        },
        Object {
          "symbol": "^GSPC",
        },
        Object {
          "symbol": "^DJI",
        },
        Object {
          "symbol": "CL=F",
        },
        Object {
          "symbol": "GC=F",
        },
        Object {
          "symbol": "^N225",
        },
        Object {
          "symbol": "^HSI",
        },
        Object {
          "symbol": "^GDAXI",
        },
        Object {
          "symbol": "^FCHI",
        },
      ],
    },
  },
}
`;
